### DISCUSSION TITLE
Request: plenty of functions are not recorded or recorded correctly with timer

### DISCUSSION THREAD

**Discussion by kirk0830** [deepmodeling/abacus-develop#5583] [General]
### Details

For any development, benchmarking is usually needed. However, there are plenty of functions left un-recorded, so the analysis on performance hotspot may be biased.

To proceed, I suggest a series of procedures to do:
1. prepare a doc illustrating the design of timer identifier which is capable for recording (at least partially) the calling relationship, This is very crucial for disinguishing the time-consuming callee and caller.
2. Insert timer::tick as many as possible, also keep in mind not to introduce redandunt dependency
The second step can be done in series of PRs rather than only one.

**Comment by mohanchen** [comment]
Excellent suggestion, we will move this to discussion.
